### PR TITLE
Settings indexLanguages

### LINKED ISSUE #102: New setting: indexLanguages array of strings
## Summary

Implement the new `indexLanguages` setting which is an array of strings.
Currently, only `["ja"]` is supported by the engine. Changes to be made:

 - Add the setting to the Settings structure/class, if any
 - Set the `indexLanguages` field in the settings integration test if the CTS
   is implemented

## References

 - Specs: https://github.com/algolia/algoliasearch-client-specs-internal/issues/40
 - Go implementation: https://github.com/algolia/algoliasearch-client-go/pull/532

### PR DESCRIPTION
**Pull request by q-litzler**
Fix #102

### CODE REVIEW

File: src/commonMain/kotlin/com/algolia/search/model/settings/Settings.kt [hunk 1]
```diff
@@ -357,11 +357,18 @@ public data class Settings(
     /**
      *  Lets you store custom data in your indices.
      */
-    @SerialName(KeyUserData) var userData: JsonObject? = null
+    @SerialName(KeyUserData) var userData: JsonObject? = null,
+    /**
+     * This parameter configures the segmentation of Japanese text at indexing time.
```
Comment (PLNech) [inline src/commonMain/kotlin/com/algolia/search/model/settings/Settings.kt]: Why not making it future proof by not tying it to japanese specifically? Considering the parameter name, it's expected more languages will be supported
```suggestion
     * This parameter configures the segmentation of text at indexing time.
```

File: src/commonMain/kotlin/com/algolia/search/model/settings/Settings.kt [hunk 2]
```diff
@@ -357,11 +357,18 @@ public data class Settings(
...
+     * Accepted value: [Language.Japanese]
+     * Input data to index is treated as Japanese text.
```
Comment (PLNech) [inline src/commonMain/kotlin/com/algolia/search/model/settings/Settings.kt]:
```suggestion
     * Input data to index is treated as the given language(s) for segmentation.
```

File: src/commonMain/kotlin/com/algolia/search/model/settings/Settings.kt [hunk 3]
```diff
@@ -357,11 +357,18 @@ public data class Settings(
...
+     */
+    @SerialName(KeyIndexLanguages) var indexLanguages: List<Language>? = null
```
Comment (PLNech) [inline src/commonMain/kotlin/com/algolia/search/model/settings/Settings.kt]: Do we want to limit it to supported language? If yes, we should create `IndexLanguages` as a subset of `Languages`

Comment (aseure) [reply src/commonMain/kotlin/com/algolia/search/model/settings/Settings.kt]: Up to you guys but looking a bit in the future, it seems like more maintenance on your side for nothing (IIRC the error is pretty self-explanatory when the engine sees a language it doesn't support yet).

Comment (q-litzler) [reply src/commonMain/kotlin/com/algolia/search/model/settings/Settings.kt]: Yes I thought about it, but deemed the cost greater than the benefit.

**Comment by PLNech** [comment]
@aseure: From my understanding, `userData` is not modified in this PR, except for the added `,` in its line :smile: 
image

**Comment by q-litzler** [comment]
I think he was referring to the `SettingsKey` file, and rightly so ;)

**Comment by PLNech** [comment]
Ah indeed :male_detective:

**Pull request merged**